### ISSUE TITLE
mumble-server installation: Ubuntu 18.04 LTS libzeroc-ice package dependency error

### ISSUE THREAD

**Issue by Septarius** [mumble-voip/mumble#3409]
OS: Ubuntu 18.04 LTS
Frequency: Always
Issue:
Installing the mumble snapshot, 1.3.0\~2729\~g2126495\~snapshot-1\~ppa1\~bionic1 , I received the following error:
"Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 mumble-server : Depends: libzeroc-ice3.6 but it is not installable
E: Unable to correct problems, you have held broken packages."

Resolution:
I compared it to Ubuntu's 18.04 1.2.19 mumble-server build and noticed that they have a newer dependency, libzeroc-ice3.7.

There are a few references to the patch for 1.2.19:
https://launchpad.net/ubuntu/+source/mumble/1.2.19-1ubuntu1
https://bugs.launchpad.net/ubuntu/+source/zeroc-ice/+bug/1758502
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=893604

A similar change might be required for 1.3.0

**Comment by Septarius** [comment]
I tried a local build of the latest HEAD commit 6195761d290fd3ba4296215e525da8008ddf86d2 .
So far it is working for how I use Murmur but I don't actually use the ice functionality.
It compiled with libzeroc-ice3.7 and since I can't make a PPA deb file I did the next best thing.
Given that it might not require a code patch to get it to work just a new PPA build for 18.04,

If someone has a Ubuntu 18.04 test server and wants to confirm the ice functionality.
No warranty implied.

Using an Ubuntu 18.04 LTS VM (fully updated), with these steps to manually create the install.

Build instructions from the Wiki: [https://wiki.mumble.info/wiki/BuildingLinux](https://wiki.mumble.info/wiki/BuildingLinux)
Setting up dev environment
```
sudo apt-get install build-essential pkg-config qt5-default libqt5svg5-dev \
libboost-dev libasound2-dev libssl-dev \
libspeechd-dev libzeroc-ice-dev libpulse-dev \
libcap-dev libprotobuf-dev protobuf-compiler \
libogg-dev libavahi-compat-libdnssd-dev libsndfile1-dev \
libg15daemon-client-dev libxi-dev 
```
Find an appropriate writable folder like the home directory.

    git clone git://github.com/mumble-voip/mumble.git mumble
    cd mumble
    git submodule init
    git submodule update

    qmake -recursive main.pro CONFIG+=no-client
    make

Get a copy of the current snapshot Ubuntu PPA.

    wget https://launchpad.net/~mumble/+archive/ubuntu/snapshot/+build/14519927/+files/mumble-server_1.3.0~2729~g2126495~snapshot-1~ppa1~bionic1_amd64.deb

Extract the deb file for editing. From: [https://lists.debian.org/debian-user/2016/08/msg00853.html](https://lists.debian.org/debian-user/2016/08/msg00853.html). 

    dpkg-deb -x mumble-server_1.3.0~2729~g2126495~snapshot-1~ppa1~bionic1_amd64.deb mumble-server
    dpkg-deb -e mumble-server_1.3.0~2729~g2126495~snapshot-1~ppa1~bionic1_amd64.deb mumble-server/DEBIAN

With the following command update the version by adding +local0 at the end. Along with updating the dependency libzeroc-ice3.6 to libzeroc-ice3.7. Read more about this in the link above.

    $EDITOR mumble-server/DEBIAN/control

Copy the newly build mumurd file into the package

    cp mumble/release/murmurd mumble-server/usr/sbin
    
Update the md5sums because the binary changed. From [https://blog.serverdensity.com/how-to-create-a-debian-deb-package/](https://blog.serverdensity.com/how-to-create-a-debian-deb-package/) 
    
    cd mumble-server
    find . -type f ! -regex '.*.hg.*' ! -regex '.*?debian-binary.*' ! -regex '.*?DEBIAN.*' -printf '%P ' | xargs md5sum > DEBIAN/md5sums
    cd ..

Rebuild the deb file to be ready for install

    fakeroot dpkg-deb -b mumble-server mumble-server_1.3.0~2729~g2126495~snapshot-1~ppa1~bionic1+local0_amd64.deb

On the test 18.04 server you want to install it on:

    sudo apt install ./mumble-server_1.3.0~2729~g2126495~snapshot-1~ppa1~bionic1+local0_amd64.deb

**Comment by C0rn3j** [comment]
Isn't this a duplicate of #2865?

**Comment by Septarius** [comment]
I believe that this is a more specific case. The change log for libzeroice3.6 in bionic it was removed from the repository on 4/7/2018 [https://launchpad.net/ubuntu/bionic/amd64/libzeroc-ice3.6](https://launchpad.net/ubuntu/bionic/amd64/libzeroc-ice3.6) and libzeroc-ice3.7 was added on the same date [https://launchpad.net/ubuntu/bionic/amd64/libzeroc-ice3.7](https://launchpad.net/ubuntu/bionic/amd64/libzeroc-ice3.7)
The last snapshot build for mumble 1.3.0 was 4/2/2018, so at the time 3.6 was the latest version available.

Though if libzeroc-ice 3.7 does have a compatibility issue even though it appears to compile fine it could be related to #2865 and/or #2728. Since I don’t use ice nor have anyone I know that does I can’t be certain on that point.

**Comment by davidebeatrici** [comment]
Patch not needed for `master`:

https://github.com/mumble-voip/mumble/blob/e54f60f44dd61c33309b97e687c1e50751c822a9/src/murmur/murmur_ice/murmur_ice.pro#L14-L19

https://github.com/mumble-voip/mumble/blob/e54f60f44dd61c33309b97e687c1e50751c822a9/src/murmur/murmur.pro#L90-L104

**Comment by davidebeatrici** [comment]
I'm closing this issue in favor of #3433, as once our PPA is working again you'll be able to install `mumble-server` from it again.

**Comment by Kissaki** [comment]
This is about the PPA snapshot packages? They are now fixed for 18.04 bionic.
